common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -503,18 +515,34 @@ def dispatch_to_register():
         """Redirects to the registration page."""
         return redirect(AUTH_DISPATCH_URLS[AUTH_ENTRY_REGISTER])
 
-    def should_force_account_creation():
-        """ For some third party providers, we auto-create user accounts """
-        current_provider = provider.Registry.get_from_pipeline({'backend': backend.name, 'kwargs': kwargs})
+    def get_provider():
+        """
+        Get's third-party provider for request
+        """
+        return provider.Registry.get_from_pipeline({'backend': backend.name, 'kwargs': kwargs})
+
+    def should_autoprovision_account():
+        """ For some third party providers we trust the provider so much that we automatically provision the account """
+        current_provider = get_provider()
+        return current_provider and current_provider.autoprovision_account
+
+    def autosubmit_registration_form():
+        """ For some third party providers, we auto-submit registration forms """
+        current_provider = get_provider()
         return current_provider and current_provider.skip_email_verification
 
     if not user:
         if auth_entry in [AUTH_ENTRY_LOGIN_API, AUTH_ENTRY_REGISTER_API]:
             return HttpResponseBadRequest()
         elif auth_entry in [AUTH_ENTRY_LOGIN, AUTH_ENTRY_LOGIN_2]:
+            if should_autoprovision_account():
+                # User has authenticated with the third party provider and provider is configured
+                # to automatically provision edX account, which is done via strategy.create_user in next pipeline step
+                return {'autoprovision': True}
```

Comment:
I think we need to move this `if` statement to come immediately after the `if not user` check. Because right now `should_autoprovision_account` is only working from the login page. If I start on the *Register* page, then when I "Sign up with TestShib" it just says "You've successfully signed in with TestShib.
We just need a little more information before you start learning with Your Platform Name Here.". It does not auto-provision the account as I would expect.